platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineService.java

```diff
@@ -362,6 +363,39 @@ public String getFileContent(@PathParam("machineId") String machineId,
         return machineManager.getMachine(machineId).readFileContent(path, startFrom, limit);
     }
 
+    /**
+     * Copies files between specified machines
+     *
+     * @param sourceMachineId
+     *         id of source machine
+     * @param sourcePath
+     *         path to resources in source machine
+     * @param targetMachineId
+     *         id of target machine
+     * @param targetPath
+     *         destination path in target machine
+     * @param noOverwrite
+     *         If "1", "true", or "True" then it will be an error
+     *         if unpacking the given content would cause an existing directory to be replaced
+     *         with a non-directory and vice versa.
+     * @throws MachineException
+     *      if any error occurs with coping resources between machines
+     */
+    @POST
+    @Path("/{machineId}/filepath/{sourcePath:.*}")
+    @RolesAllowed("user")
+    public void transferFilesBetweenMachines(@PathParam("machineId") String sourceMachineId,
+                                             @PathParam("sourcePath") String sourcePath,
+                                             @QueryParam("targetMachineId") String targetMachineId,
+                                             @QueryParam("targetPath") String targetPath,
+                                             @DefaultValue("false") @QueryParam("noOverwrite") Boolean noOverwrite)
+            throws NotFoundException, ServerException, ForbiddenException {
+        checkCurrentUserPermissions(machineManager.getMachine(sourceMachineId));
+
+        machineManager.getMachine(targetMachineId)
```

Comment:
With this code you declare copying via API instance. What if 
`machineManager.getMachine(machineId).copy(targetMachineId, targetPath, overwrite)`
@akorneta  @skabashnyuk WDYT?